velox/exec/Operator.h

```diff
@@ -476,6 +476,16 @@ class Operator : public BaseRuntimeStatWriter {
   // 'identityProjections_' and 'resultProjections_'.
   RowVectorPtr fillOutput(vector_size_t size, BufferPtr mapping);
 
+  // Returns the max number of rows in the output batch. This uses
+  // estimateOutputRowSize to calculate how many rows fit in
+  // preferredOutputBatchBytes. It caps the number of rows at 10K and returns at
+  // least one row. If the operator does not have estimateOutputRowSize, returns
+  // preferredOutputBatchRows.
+  uint32_t outputBatchRows() const;
+
+  // Returns an estimate of the output row size if an operator has an estimate.
+  virtual std::optional<uint64_t> estimateOutputRowSize() const;
```

Comment:
I was thinking about this a bit more and wondering if we can avoid introducing a new virtual method, but rather pass the estimate to outputBatchRows():

```
uint32_t outputBatchRows(std::optional<uint64_t> estimatedRowSize = std::nullopt) const;
```

Reply:
I got your idea. Revised.